components/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/handlers/security/saml2/SAML2AuthenticationHandler.java

```diff
@@ -114,6 +152,8 @@ public void init(SynapseEnvironment synapseEnvironment) {
         try {
             authenticator = new OAuthAuthenticator();
             saml2Authenticator = new SAML2Authenticator();
+            webApp = new WebApp();
+            initJWTCapabilities();
         } catch (Exception e) {
             // Just throw it here - Synapse will handle it
             throw new SynapseException("Error while initializing SAML or OAuth authenticator");
```

Comment:
As RuwanA commented in above we let the system initialize even with partial errors and then throws a SynapseException. That is why we are having generic exception catch block. I will add some comments in the code mentioning the reason to have generic exception catch block.